autoload/lsp/omni.vim

```diff
@@ -248,8 +248,10 @@ function! lsp#omni#default_get_vim_completion_item(item, ...) abort
     endif
 
     if has_key(a:item, 'documentation')
-        if type(a:item['documentation']) == type('')
+        if type(a:item['documentation']) == type('') " field is string
             let l:completion['info'] .= a:item['documentation']
+        elseif a:item['documentation']['kind'] == 'plaintext' " field is MarkupContent
```

Comment:
Should check for type as well: `type(a:item['documentation']) == type({})`.
Are we going to do something about Markdown as well?

Reply:
What is the purpose of the check? That it's non-empty? Maybe try to catch that first?

Re Markdown: How would we do that? If it's just formatted text, we can pass that through (in which case we don't need the check), but if it's an embedded image, I don't know how this could be handled in nvim...

Reply:
If documentation is neither a string, nor a dict, then the subscript `['kind']` will fail (could be the case if the server returns null for example).

For markdown: I'm just thinking that maybe some servers return some text, but wrap everything in markdown, because in VSCode it won't matter.

Reply:
True, but the server should check for Markdown capabilities first (texlab does, at least for hover).

But the worst that can happen is a preview window full of an ascii-coded image link ;)

Reply:
Ah right, that's another separate capability. I always forget about those :(
Honestly, I'm fine with it either way: whether you decide to check for `plaintext` or not.

An ascii coded image does sound amazing tho :p

Reply:
Let's be permissive; it's always easier to open (and act on) an issue complaining about weird text in the preview than about something not showing that should.